### PR TITLE
Lightning payment history

### LINKED ISSUE #165: Payment history
Fetch Bitcoin and Lightning payment history from the wallet backend and plug it in instead of the mocked data.

This concerns the `Dashboard` where we have a combined history of Bitcoin and Lightning, and the specific Lightning and Bitcoin wallet screens.

Furthermore this payment history could serve as the basis for the CFD trading overview (which is basically a subset of the Lightning history + some additional CFD specific data).

### PR DESCRIPTION
**Pull request by bonomat**
Resolves #165 

Note: 
* What is missing is the overview of the actual CFDs. I think we don't have this data yet.
* We cannot know easily which transaction opened the channel. For that we will need to store some additional data somewhere.

### CODE REVIEW

**Review by klochowicz — APPROVED** [review]
LGTM!

**Comment by da-kami** [comment]
> Resolves #165
> 
> Note:
> 
>     * What is missing is the overview of the actual CFDs. I think we don't have this data yet.

This is tracked in https://github.com/itchysats/10101/issues/201
It will be enabled by: https://github.com/itchysats/10101/pull/220

>     * We cannot know easily which transaction opened the channel. For that we will need to store some additional data somewhere.

Yeah, we would have to store metadata in the database for that as well! I added https://github.com/itchysats/10101/issues/226 for this task - feel free to extend it if the description is not complete! 
It does not have the highest prio for me, but I think we can easily add this feature once we have the database (https://github.com/itchysats/10101/pull/220)

File: rust/src/wallet.rs [hunk 1]
```diff
@@ -228,6 +230,41 @@ pub fn get_bitcoin_tx_history() -> Result<Vec<bdk::TransactionDetails>> {
...
+            status: payment_info.status.clone().into(),
+            timestamp: payment_info.timestamp,
+        })
+        .collect();
+
+    let mut inbound = inbound
+        .iter()
+        .map(|(_, payment_info)| LightningTransaction {
+            tx_type: LightningTransactionType::Payment,
+            flow: Flow::Outbound,
```
Comment (da-kami) [inline rust/src/wallet.rs:257]:
```suggestion
            flow: Flow::Inbound,
```

🛠️ I think this is a copy-paster that should be fixed :D

**Review by da-kami — APPROVED** [review]
Noice, but I think there is a copy-paster that needs a fix, see comment :)

A screenshot would be great :)

File: rust/src/wallet.rs (hunk 1 above)
Comment (bonomat) [reply rust/src/wallet.rs]: Good catch. Fixed

**Comment by bonomat** [comment]
> A screenshot would be great :)

Didn't change anything in the layout, so the old mocks would have shown the same :) 

bors r+

**Comment by da-kami** [comment]
> > A screenshot would be great :)
> 
> Didn't change anything in the layout, so the old mocks would have shown the same :)
> 
> bors r+

Sure, but it's real data now and not mocked garbage :D

**Comment by bonomat** [comment]
> Sure, but it's real data now and not mocked garbage :D

you wouldn't know the difference because the mocked data was such high quality and accurately 😅

**Pull request merged**